docs/how-to-guides/shared-libraries.rst

```diff
@@ -0,0 +1,253 @@
+.. SPDX-FileCopyrightText: 2024 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _shared-libraries:
+
+**********************
+Using shared libraries
+**********************
+
+Python projects may build shared libraries as part of their project, or link
+with shared libraries from a dependency. This tends to be a common source of
+issues, hence this page aims to explain how to include shared libraries in
+wheels, any limitations and gotchas, and how support is implemented in
+``meson-python`` under the hood.
+
+We distinguish between *internal* shared libraries, i.e. they're built as part
+of the build executed by ``meson-python``, and *external* shared libraries that
+are only linked against from targets (usually Python extension modules) built
+by ``meson-python``. For internal shared libraries, we also distinguish whether
```

Comment:
I find the distinction between internal and external shared libraries a bit hard to read. What about something like:

We distinguish between *internal* shared libraries that are built as part of the project, and *external* shared libraries provided by project dependencies and that are linked with the project build artifacts.

or something like this.

Reply:
That sounds better, adopted.